compiler/AST/build.cpp

```diff
@@ -162,16 +163,33 @@ BlockStmt* buildPragmaStmt(Vec<const char*>* pragmas,
   return stmt;
 }
 
-static Expr* convertStringLiteral(Expr *e) {
-  if (SymExpr* s = toSymExpr(e)) {
-    VarSymbol *v = toVarSymbol(s->var);
-    INT_ASSERT(v);
-    if (v->immediate &&
-        v->immediate->const_kind==CONST_KIND_STRING) {
-      return new CallExpr("toString", s);
+
+//
+// A helper function that's useful in a few places in this file to
+// conditionally convert an Expr to a 'const char*' in the event that
+// it's an immediate string.
+//
+static const char* toImmediateString(Expr* expr) {
+  if (SymExpr* se = toSymExpr(expr)) {
+    if (VarSymbol* var = toVarSymbol(se->var)) {
+      if (var->isImmediate()) {
+        Immediate* imm = var->immediate;
+        if (imm->const_kind == CONST_KIND_STRING) {
+          return imm->v_string;
+        }
+      }
     }
   }
-  return e;
+  return NULL;
+}
+
+
+static Expr* convertStringLiteral(Expr *e) {
+  if (toImmediateString(e) != NULL) {
+    return new CallExpr("toString", e);
```

Comment:
Just want to verify that the `toString` primitive doesn't care what type `e` is. Previously it was cast to a symExpr, now it will just be an Expr